step-plans/step-plans-core/src/main/java/step/core/artefacts/reports/aggregated/AggregatedReportViewBuilder.java

```diff
@@ -35,8 +35,14 @@ public AggregatedReportView buildAggregatedReportView(AggregatedReportViewReques
         Objects.requireNonNull(request);
         Execution execution = executionAccessor.get(executionId);
         String aggregatedReportRoot = execution.getResolvedPlanRootNodeId();
-        ResolvedPlanNode resolvedPlanNode = resolvedPlanNodeAccessor.get(aggregatedReportRoot);
-        return recursivelyBuildAggregatedReportTree(resolvedPlanNode, request);
+        //Make sure the resolved Plan is available
```

Comment:
Looks correct. Here's an equivalent solution that avoids explicit branches. I'll leave it up to you to decide which one you prefer and find more readable, I'd be fine with both.
```
        return Optional.ofNullable(execution.getResolvedPlanRootNodeId())
                .map(resolvedPlanNodeAccessor::get)
                .map(node -> recursivelyBuildAggregatedReportTree(node, request))
                .orElse(null);
```

Reply:
feedback implemented